presto-orc/src/main/java/com/facebook/presto/orc/reader/BooleanSelectiveStreamReader.java

```diff
@@ -160,7 +165,13 @@ public int read(int offset, int[] positions, int positionCount)
 
         allNulls = false;
 
-        if (outputRequired) {
+        if (useBatchMode()) {
+            // values need to be totalPositionCount,
```

Comment:
Minor: would be nice to reword and reformat this comment. ByteSelectiveStreamReader has a nice wording for this comment.